nekoyume/Assets/_Scripts/UI/Widget/Popup/ItemMaterialSelectPopup.cs

```diff
@@ -0,0 +1,119 @@
+        [SerializeField] private TextMeshProUGUI itemNameText;
+        [SerializeField] private ConditionalButton combineButton;
+        [SerializeField] private TextMeshProUGUI selectCountText;
+        [SerializeField] private TextMeshProUGUI requiredCountText;
+        [SerializeField] private ItemMaterialSelectScroll scroll;
+        [SerializeField] private Button closeButton;
+
+        private List<ItemMaterialSelectScroll.Model> Models { get; set; }
+
+        private readonly List<IDisposable> _disposablesForShow = new();
+        private readonly Dictionary<int, Dictionary<int, int>> _selectedMaterialsByRecipe = new();
+
+        protected override void Awake()
+        {
+            base.Awake();
+
+            closeButton.onClick.AddListener(() =>
+            {
+                Close();
+                AudioController.PlayClick();
+            });
+        }
+
+        public void Show(
+            EventMaterialItemRecipeSheet.Row recipeRow,
+            Action<Dictionary<int, int>> onSubmit)
+        {
+            combineButton.Interactable = false;
+            _disposablesForShow.DisposeAllAndClear();
+
+            var material = ItemFactory.CreateMaterial(
+                Game.Game.instance.TableSheets.MaterialItemSheet,
+                recipeRow.ResultMaterialItemId
+            );
+            var resultItemCount = recipeRow.ResultMaterialItemCount;
+            var resultCountableItem = new CountableItem(material, resultItemCount);
+            itemView.SetData(resultCountableItem);
+            itemNameText.text = Game.Game.instance.TableSheets
+                .ItemSheet[recipeRow.ResultMaterialItemId].GetLocalizedName(false, false);
+
+            var requiredItemCount = recipeRow.RequiredMaterialsCount;
+            selectCountText.text = "0";
+            requiredCountText.text = $"/{requiredItemCount}";
+
+            var inventoryItems = States.Instance.CurrentAvatarState.inventory.Items;
+            var lastSelectedMaterials = _selectedMaterialsByRecipe.GetValueOrDefault(recipeRow.Id);
+            var models = recipeRow.RequiredMaterialsId
+                .Select(id =>
+                {
+                    var itemBase = ItemFactory.CreateMaterial(Game.Game.instance.TableSheets.MaterialItemSheet[id]);
+                    var itemCount = inventoryItems.FirstOrDefault(item => Equals(item.item.Id, itemBase.Id))?.count ?? 0;
+                    var countableItem = new CountableItem(itemBase, itemCount);
+                    var lastSelectedCount = lastSelectedMaterials?.GetValueOrDefault(id) ?? 0;
+                    return new ItemMaterialSelectScroll.Model(countableItem, Mathf.Min(lastSelectedCount, itemCount));
+                })
+                .OrderByDescending(model => model.Item.Count.Value)
+                .ThenBy(model => model.Item.ItemBase.Value.Id)
+                .ToList();
+            Models = models;
+            ObservableExtensions.Subscribe(Observable.NextFrame(), _ =>
```

Comment:
If you using `using UniRx;` inside of namespace `Nekoyume.UI`, This line can be using as extension method.